### PR TITLE
Dynamic Shortable Asset List

### PR DESCRIPTION
**Pull request by kethcode**
##### useGetShortableSynths.ts
Added a new query. Updated ethcall to 4.3.6 to enable OP support.  L2 Synths are retrieved from synthetixjs using multicall, memoed into each file that uses it.  L1 list is hard coded into the query since they are deprecated, and the L1 contract gives us no method to retrieve them dynamically. 

##### ShortingStats.tsx
Replaced SYNTHS_TO_SHORT and SYNTHS_TO_SHORT_L1 with a single query that populates SYNTHS_TO_SHORT, and updated hooked memos.

##### useFeeReclaimPeriodsQuery.ts
new version of ethcall errors when passed {} parameter. It also returns unknown[] instead of [any,any,any], like the previous version did, so I cast the result.

##### useRedeemableDeprecatedSynthsQuery.ts
new version of ethcall errors when passed {} parameter

##### useTokensBalancesQuery.ts
new version of ethcall errors when passed {} parameter

##### Resolved Various Lint Errors that somehow made it into dev (came down with my upstream fetch)

```
/home/kethic/archive/kwenta/components/TVChart/TVChart.tsx
  85:5  warning  React Hook React.useEffect has missing dependencies: 'autosize', 'containerId', 'fullscreen', 'interval', 'libraryPath', 'overrides', and 'studiesOverrides'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```

```
/home/kethic/archive/kwenta/sections/exchange/TradeCard/Cards/CombinedMarketDetailsCard.tsx
   2:25  warning  'useMemo' is defined but never used                                                                                         @typescript-eslint/no-unused-vars
  50:5   warning  React Hook useEffect has a missing dependency: 'rates24hQuery.isSuccess'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
```

```
/home/kethic/archive/kwenta/sections/exchange/hooks/useExchange.tsx
  541:3  warning  React Hook useMemo has an unnecessary dependency: 'isL2'. Either exclude it or remove the dependency array  react-hooks/exhaustive-deps
```

## Related issue
https://github.com/Kwenta/kwenta/issues/359

## Motivation and Context
Allow for automated deployment of new synthetix synths to kwenta shorting market

##### ShortingHistory.tsx
The right-side drop-down populates with the correct L2 synths. I opened and closed positions on every L2 synth, confirming it was selectable, and showed up correctly under Your Positions.

##### ShortingReward.tsx
This is only used on L1.  Verified the synth reward list is correct for L1 after the change.

##### ShortingStats.tsx
This is used on L1 and L2.  Verified the synth list displayed is correct on both L1 and L2 after the change.

##### useFeeReclaimPeriodsQuery.ts
I don't know how to test this one other than the console.log commands I used during dev.

##### useRedeemableDeprecatedSynthsQuery.ts
DashboardCard uses this.  Synth list appears to still work after the change.

##### useTokensBalancesQuery.ts
This is used in useExchange and SelectTokenModal.  Both of these appear to still work after the change.

### CODE REVIEW

**Comment by kethcode** [comment]
I don't know what to do to resolve these checks.  they never pass, and it concerns me.

File: components/TVChart/TVChart.tsx [hunk 1]
```diff
@@ -82,7 +82,17 @@ export function TVChart({
...
+	}, [
+		baseCurrencyKey,
+		quoteCurrencyKey,
+		autosize,
+		containerId,
+		fullscreen,
+		interval,
+		libraryPath,
+		overrides,
+		studiesOverrides,
```
Comment (koredefashokun) [inline components/TVChart/TVChart.tsx:94]: Adding these causes some flickering on the `TVChart` component when it is refocused.

File: queries/synths/useFeeReclaimPeriodsQuery.ts [hunk 2]
```diff
@@ -50,10 +50,9 @@ const useFeeReclaimPeriodsQuery = (
 			}
 
 			const waitingPeriodsRaw = (await ethcallProvider.all(
-				waitingPeriodCalls,
-				{}
+				waitingPeriodCalls
 			)) as ethers.BigNumberish[];
-			const feesRaw = await ethcallProvider.all(feeCalls, {});
+			const feesRaw = (await ethcallProvider.all(feeCalls)) as [any, any, any];
```
Comment (koredefashokun) [inline queries/synths/useFeeReclaimPeriodsQuery.ts:55]: Not sure if this type is correct. `settlementOwing` returns a tuple of three `uint` values. I think this should be:

```suggestion
			const feesRaw = (await ethcallProvider.all(feeCalls)) as [ethers.BigNumberish, ethers.BigNumberish, ethers.BigNumberish][];
```

**Comment by koredefashokun** [comment]
@kethcode Sorry for the delayed review. Left a few comments above.

**Comment by kethcode** [comment]
Thank you for the review. I will make these changes

File: queries/synths/useFeeReclaimPeriodsQuery.ts (hunk 2 above)
Comment (kethcode) [reply queries/synths/useFeeReclaimPeriodsQuery.ts:55]: > Not sure if this type is correct. `settlementOwing` returns a tuple of three `uint` values. I think this should be:

this does not work.  tried it, and sub requires a number type, and throws an error on ethers.BigNumberish.  I understand the desire to type this, but whatever is coming back needs iterable, so it's not a Number either.  We can create a ticket to fix this later, if we want.  I've added a TODO for now

File: components/TVChart/TVChart.tsx (hunk 1 above)
Comment (kethcode) [reply components/TVChart/TVChart.tsx:94]: I've removed these and added // eslint-disable-next-line react-hooks/exhaustive-deps

File: components/TVChart/TVChart.tsx [hunk 3]
```diff
@@ -82,17 +82,8 @@ export function TVChart({
...
-		autosize,
-		containerId,
-		fullscreen,
-		interval,
-		libraryPath,
-		overrides,
-		studiesOverrides,
-	]);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
```
Comment (koredefashokun) [inline components/TVChart/TVChart.tsx:86]: I think we need the `baseCurrencyKey` and `quoteCurrencyKey` in the deps array here (only these two). Sorry for the confusion.

File: queries/synths/useFeeReclaimPeriodsQuery.ts [hunk 4]
```diff
@@ -52,7 +52,7 @@ const useFeeReclaimPeriodsQuery = (
 			const waitingPeriodsRaw = (await ethcallProvider.all(
 				waitingPeriodCalls
 			)) as ethers.BigNumberish[];
-			const feesRaw = (await ethcallProvider.all(feeCalls)) as [any, any, any];
+			const feesRaw = (await ethcallProvider.all(feeCalls)) as [any, any, any]; // TODO: replace any with correct iterable type
```
Comment (koredefashokun) [inline queries/synths/useFeeReclaimPeriodsQuery.ts:55]: Sorry about the confusion here as well. Tested it locally and it should be:
```suggestion
			const feesRaw = (await ethcallProvider.all(feeCalls)) as [ethers.BigNumber, ethers.BigNumber, ethers.BigNumber][];
```

**Comment by koredefashokun** [comment]
Also, it looks like there are conflicts in the package-lock.json. I think you should be able to fix them by deleting the file and re-running `npm i`. Only caveat is to make sure you're on npm v7.

**Comment by kethcode** [comment]
@koredefashokun I saw that.  However, I need to upgrade ethcall to 4.3.6, so the changes need to occur.  Someone else just pushed a lock file with 3.2.0 in it since I submit the PR.  Should I pull the latest dev v1, merge and recommit?

File: queries/synths/useFeeReclaimPeriodsQuery.ts (hunk 4 above)
Comment (kethcode) [reply queries/synths/useFeeReclaimPeriodsQuery.ts:55]: will fix

File: components/TVChart/TVChart.tsx (hunk 3 above)
Comment (kethcode) [reply components/TVChart/TVChart.tsx:86]: will fix

**Comment by kethcode** [comment]
@koredefashokun @platschi should be done now.

**Comment by kethcode** [comment]
I give up.  Turning off my linter, deleting this PR, and I'll resubmit with a narrower scope.

**Pull request closed without merging**